spec/lib/hubspot/company_properties_spec.rb

```diff
@@ -14,158 +14,325 @@
+      context "when included groups are provided" do
+        it "returns the specified groups" do
+          VCR.use_cassette("company_properties/groups_included") do
+            group_a = "socialmediainformation"
+            group_b = "companyinformation"
+
+            Hubspot::CompanyProperties.create!({
+              name: "instagram_handle",
+              label: "Instagram Handle",
+              description: "The company's Instagram handle.",
+              groupName: group_a,
+              type: "string",
+              fieldType: "text"
+            })
+            Hubspot::CompanyProperties.create!({
+              name: "fax_number",
+              label: "Fax Number",
+              description: "The company fax number.",
+              groupName: group_b,
+              type: "string",
+              fieldType: "text"
+            })
+
+            response = Hubspot::CompanyProperties.groups(
+              {},
+              { include: group_a }
+            )
+
+            assert_hubspot_api_request(:get, "/companies/v2/groups")
+
+            Hubspot::CompanyProperties.delete!("instagram_handle")
+            Hubspot::CompanyProperties.delete!("fax_number")
+
+            group_names = response.map { |group| group["name"] }
+            expect(group_names).to include(group_a)
+            expect(group_names).not_to include(group_b)
+          end
+        end
+      end
 
-        it 'should return groups that were not excluded' do
-          response = Hubspot::CompanyProperties.groups({}, { exclude: groups })
-          response.each { |p| expect(groups.include?(p['name'])).to be false }
+      context "when excluded groups are provided" do
+        it "returns groups that were not excluded" do
+          VCR.use_cassette("company_properties/groups_excluded") do
+            group_a = "socialmediainformation"
+            group_b = "companyinformation"
+
+            Hubspot::CompanyProperties.create!({
+              name: "instagram_handle",
+              label: "Instagram Handle",
+              description: "The company's Instagram handle.",
+              groupName: group_a,
+              type: "string",
+              fieldType: "text"
+            })
+            Hubspot::CompanyProperties.create!({
+              name: "fax_number",
+              label: "Fax Number",
+              description: "The company fax number.",
+              groupName: group_b,
+              type: "string",
+              fieldType: "text"
+            })
+
+            response = Hubspot::CompanyProperties.groups(
+              {},
+              { exclude: group_a }
+            )
+
+            assert_hubspot_api_request(:get, "/companies/v2/groups")
+
+            group_names = response.map { |group| group["name"] }
+            expect(group_names).not_to include(group_a)
+            expect(group_names).to include(group_b)
+
+            Hubspot::CompanyProperties.delete!("instagram_handle")
```

Comment:
Will these get called if the test fails? Should these be in an after hook or something similar to ensure they happen?

Reply:
I left an answer to a similar question here: https://github.com/adimichele/hubspot-ruby/pull/159#discussion_r241551906

Thinking about this a bit more, I could do something that calls these lines and rescues in the event the call fails. 

The benefit is the test author doesn't have to use a tool, like Postman, to delete the created record before they can re-run the test.

The downside to this change is the failure to delete a record is hidden. That's probably fine but it means a test author could commit a VCR recording where the `delete` action is a non-200 response. This could lead to some confusion when attempting to update this test's recording and the test errors due to a validation error. As we move away from using the "demo" account, that will become less of a likely scenario.